source/AI.cpp

```diff
@@ -4765,19 +4777,18 @@ void AI::UpdateStrengths(map<const Government *, int64_t> &strength, const Syste
 	}
 
 	// Ships with nearby allies consider their allies' strength as well as their own.
-	for(const auto &it : ships)
+	mutex lock;
+	for_each(parallel::par, ships.begin(), ships.end(), [&](const auto &it)
```

Comment:
same for this one, does not make a whole lot of sense to have small stuff like this in different threads

Reply:
It does. If you look at the code, this potentially iterates over every single ship, times every friendly government. That becomes quite a lot of operations, even if the code looks small.

Reply:
ok, so amount of ships, say 500(way too much), every friendly government, say 50, so we get to 25000 iterations, modern cpus blast away much higher numbers in a still miniscule time